### PR TITLE
Add documentation for 'setup.cfg' configuration

### PR DESCRIPTION
**Pull request by stephenfin**
An issue with `[egg_info]` was seen with https://github.com/stephenfin/rst2txt. Take the opportunity to highlight this issue along with general `setup.cfg` support.

### CODE REVIEW

**Review by RonnyPfannschmidt — CHANGES_REQUESTED** [review]
great job, the markup enhancements are neat and the added informations realy drive home a more metadata oriented approach

for it to be fully embracing of the setuptools features we also need to show people the patterns needed for minimal setuptools versions, else they are in for bad surprises on enterprise distro style setups

**Comment by stephenfin** [comment]
> for it to be fully embracing of the setuptools features we also need to show people the patterns needed for minimal setuptools versions, else they are in for bad surprises on enterprise distro style setups

Good call. Done.

**Comment by anthrotype** [comment]
if you add setuptools_scm to the build requirements in pyproject.toml, then you could do away with setup_requires and eggs altogether, no?

**Comment by anthrotype** [comment]
maybe for the time being would be better to recommend to add setuptools_scm to both pyproject.toml and the setup_requires, so that newer pips will use the former (and skip the latter as already resolved by the time setup_requires are evaluated), and older pips that don't know about pyproject.toml can keep using setup_requires (provided they don't fail with some obscure SSL error).

**Comment by stephenfin** [comment]
> maybe for the time being would be better to recommend to add setuptools_scm to both pyproject.toml and the setup_requires, so that newer pips will use the former (and skip the latter as already resolved by the time setup_requires are evaluated), and older pips that don't know about pyproject.toml can keep using setup_requires (provided they don't fail with some obscure SSL error).

Done

**Review by RonnyPfannschmidt — APPROVED** [review]
great work, i'm going to merge later this evening, please give me a note if you wish to streamline history a bit

**Comment by stephenfin** [comment]
Yup, happy for you to squash the last three commits. I just kept them separate because I've had people complain about force pushing on GitHub.

**Comment by RonnyPfannschmidt** [comment]
@stephenfin my personal stance is

* i keep commits separate while the pr is being developed
* i will squash my own commits after the review process is done and i never squash commits of others
* i give contributors the opportunity to squash themselves before i merge

**Comment by stephenfin** [comment]
Oh, that all makes sense. In this particular case though, I'm away from a computer and won't be near one again until next week. As such, if you're happy to squash them, go for it.

File: README.rst [hunk 1]
```diff
@@ -1,61 +1,119 @@
...
+
+    [options]
+    setup_requires =
+      setuptools_scm
+    ...
+
+.. important::
+
+    Ensure neither the ``[metadata] version`` option nor the ``[egg_info]``
+    section are not defined, as these will interfere with ``setuptools_scm``.
```
Comment (marcelm) [inline README.rst:103]: The extra “not” in this line should probably be removed

**Comment by marcelm** [comment]
This PR would close #315

**Comment by RonnyPfannschmidt** [comment]
good reminder, it got lost over the holiday season, i'll put it in my week todo

**Pull request merged**